src/Impute.jl

```diff
@@ -1,14 +1,38 @@
 module Impute
 
-using DataFrames
+using IterTools
 using Statistics
+using StatsBase
+using Tables: Tables, materializer, istable
 
-import DataFrames: DataFrameRow
-import Base.Iterators
+import Base.Iterators: drop
 
 export impute, impute!, chain, chain!, drop, drop!, interp, interp!, ImputeError
 
-const Dataset = Union{AbstractArray, DataFrame}
+function __init__()
+    sym = join(["chain", "chain!", "drop", "drop!", "interp", "interp!"], ", ", " and ")
+
+    @warn(
+        """
+        The following symbols will not be exported in future releases: $sym.
+        Please qualify your calls with `Impute.<method>(...)` or explicitly import the symbol.
+        """
+    )
+
+    @warn(
+        """
+        The default limit for all impute functions will be 1.0 going forward.
+        If you depend on a specific threshold please pass in an appropriate `AbstractContext`.
+        """
+    )
+
+    @warn(
+        """
+        All matrix imputation methods will be switching to the JuliaStats column-major convention
```

Comment:
This is a lie, nothing is consistent in JuliaStats.
```suggestion
        All matrix imputation methods will be switching to the column-major convention
```

Reply:
Hmmm, I should probably update the MultivariateStats.jl docs then 🤔

Reply:
I don't have an example at hand.
On a quicklook it seems MultivariateStats.jl and Distrubutions.jl are col-obs,
I feel like i used to have one.
Maybe I am wrong, and we can have nice things...

Reply:
Unfortunately, MultivariateStats and Distributions use columns as observations, but Statistics (e.g. `cor`) and StatsBase use rows as observations, and so do Tables.jl, DataFrames and StatsModels. So the only safe solution for now is to require explicitly specifying which layout you are using (like in Distances).

Reply:
I'd rather just pick a convention for the entire package an stick to that. I think if it's documented then that's fine.